### PR TITLE
[ADD] stock_production_lot_firmware_version

### PR DESCRIPTION
**Pull request by bodedra**
#321

### CODE REVIEW

**Review by max3903 — CHANGES_REQUESTED** [review]
@bodedra Please use the OCA logo from https://github.com/OCA/maintainer-tools/blob/master/template/module/static/description/icon.png

**Review by max3903 — APPROVED** [review]

File: stock_production_lot_firmware_version/views/stock_view.xml [hunk 1]
```diff
@@ -0,0 +1,29 @@
...
+    </record>
+    <!-- Lots/Serial Numbers search view -->
+    <record id="search_product_lot_filter_fm_version" model="ir.ui.view">
+        <field name="name">Production Lots Filter</field>
+        <field name="model">stock.production.lot</field>
+        <field name="inherit_id" ref="stock.search_product_lot_filter"/>
+        <field name="arch" type="xml">
+            <data>
+                <field name="name" position="attributes">
+                    <attribute name="filter_domain">['|','|',('name','ilike',self),('ref','ilike',self),('firmware_version','ilike',self)]</attribute>
```
Comment (guewen) [inline stock_production_lot_firmware_version/views/stock_view.xml]: As soon as another addon will have the same idea about adding a field in this domain, only one of them will work. Wouldn't it be safer to add a new search filter? Or maybe customize the name_search?

The firmware version doesn't seem so important to me to be in the "name" search filter, am I wrong?

Comment (bodedra) [reply stock_production_lot_firmware_version/views/stock_view.xml]: Agree.

We have two options for these.

1. Adding new firmware version filter. (User need to select particular these filter option)

2. Override `name_search`. (only append domain filter)

What your suggestion ?

Comment (max3903) [reply stock_production_lot_firmware_version/views/stock_view.xml]: @bodedra As long as the user can search using the firmware version field in the advanced filters, we are good. No need for a specific filter, nor include it in the name_search.

**Review by guewen — APPROVED** [review]
Thanks!

**Review by yvaucher — APPROVED** [review]

**Pull request merged**